src/distribute_kv_cache/rpc/workerpool.rs

```diff
@@ -0,0 +1,223 @@
+use std::{
+    fmt::{self, Debug},
+    sync::Arc,
+};
+
+use async_trait::async_trait;
+use tracing::debug;
+
+use super::error::RpcError;
+
+/// A job that can be executed by a worker.
+type JobImpl = Box<dyn Job + Send + Sync + 'static>;
+
+/// A worker that can execute async jobs.
+#[derive(Clone)]
+pub struct WorkerPool {
+    /// The number of workers in the worker pool.
+    /// Current implementation is that the worker pool with a fixed number of workers.
+    /// TODO: Test if we need a dynamic worker pool.
+    max_workers: usize,
+    /// The maximum number of jobs that can be waiting in the job queue.
+    max_waiting_jobs: usize,
+    /// The job queue for the worker pool, with a maximum buffer capacity of `max_waiting_jobs`.
+    /// 1. When the job queue is full, the worker pool will block to receive new jobs.
+    /// 2. When sender is dropped, the receiver will return `None` and the worker pool will shutdown.
+    /// 3. Check the receiver reference count to determine if the worker pool is still alive.
+    job_sender: Arc<flume::Sender<JobImpl>>,
+    /// The workers in the worker pool.
+    worker_queue: Vec<Worker>,
+}
+
+impl Debug for WorkerPool {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("WorkerPool")
+            .field("max_workers", &self.max_workers)
+            .field("max_waiting_jobs", &self.max_waiting_jobs)
+            .field("worker_queue_len", &self.worker_queue.len())
+            .finish_non_exhaustive()
+    }
+}
+
+impl WorkerPool {
+    /// Create a new worker pool, which contains a number of max workers and max waiting jobs.
+    #[must_use]
+    pub fn new(max_workers: usize, max_waiting_jobs: usize) -> Self {
+        let (job_sender, job_receiver) = flume::bounded::<JobImpl>(max_waiting_jobs);
+        let mut worker_queue = Vec::new();
+
+        // In current implementation, we create a fixed number of workers.
+        let receiver = Arc::new(job_receiver);
```

Comment:
You don't have to wrap a Arc around. The receiver can clone by itself.

Reply:
Got, I have remove the `Arc` here.